src/redux/missions/missions.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable camelcase */
+/* eslint-disable comma-dangle */
+/* eslint-disable no-use-before-define */
+const FETCH_LOADING = 'FETCH_LOADING';
+const FETCH_SUCCESS_MISSIONS = 'FETCH_SUCCESS_MISSIONS';
+const FETCH_ERROR = 'FETCH_ERROR';
+
+const initialState = {
+  loading: true,
+  missions: [],
+  userMissions: [],
+};
+
+export const fetchPostsRequestMissions = () => async (dispatch) => {
+  dispatch(fetchPostsLoading());
+  const request = await fetch('https://api.spacexdata.com/v3/missions');
+  const result = await request.json();
+  dispatch(
+    fetchPostsSuccessMissions(
+      result.map((mission) => {
+        const selectedData = (({ mission_id, mission_name, description }) => ({
+          mission_id,
+          mission_name,
+          description,
+          join: false,
+        }))(mission);
+        return selectedData;
+      })
+    )
+  );
+};
```

Comment:
Optional: Move the Middleware functions to middleware.js and import them back where you need them.
It's better to spread the main reducer file and middleware so our code will be more readable.